WolvenKit.App/Services/AppScriptService.Hook.cs

```diff
@@ -47,57 +55,41 @@ private bool OnSaveHook(string filePath, ref CR2WFile cr2wFile)
 
         _settingsManager.ScriptStatus ??= new();
 
-        ScriptFile? script = null;
-        foreach (var scriptFile in GetScripts())
+        string? jsonContent = null;
+
+        foreach (var scriptFile in GetHookScripts(HookType.Save, extStr))
         {
-            if (scriptFile.Type != ScriptType.Hook)
+            if (_settingsManager.ScriptStatus.TryGetValue(scriptFile.Path, out var enabled) && !enabled)
```

Comment:
```suggestion
            if (!_settingsManager.ScriptStatus.TryGetValue(scriptFile.Path, out var enabled) || enabled != true)
```

Reply:
Right now, it will not continue if TryGetValue fails

Reply:
Thats true, but it was always like that.
Do you want it to be changed?

Reply:
Would make sense, yeah :)

Reply:
Then we should do it in a separate PR.
Would need some refactor, since it only gets added there if you toggle the checkbox in the scriptmanager.
Every hook would be ignored even if it appears checked in the manager.

Reply:
Wdym with "invalid headers"?
The dict stores the state of the scripts (Enabled/Disabled).
Currently it only gets added there, if the user switches it from the default state Enabled to Disabled.

Reply:
Oh - yeah, okay, that makes sense >.<  But if the headers can't be read, then the script shouldn't be enabled, should it?

Reply:
Anyhow, this was just a suggestion :) otherwise LGTM

Reply:
Ah, the wscript header.
If its invalid, `GetScripts()` / `GetHookScripts()` doesn't return.